### PR TITLE
Case 20722: Ring Gizmo Entities to replace Circle Overlays

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/20722/Add-missing-Circle-Overlay-properties-to-entities

Test plan:
- Run [this](https://gist.githubusercontent.com/SamGondelman/9589dbe96de48e2c2a963f8ef31f54d4/raw/73eec4391f2168760b482c48833d060f4335fd37/RingGizmoEntity.js), you'll see this:
image

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16707/HighFidelity-Beta-PR14764-ac513df.apk

File: libraries/entities-renderer/src/RenderableGizmoEntityItem.cpp [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    bool needsUpdate = resultWithReadLock<bool>([&] {
+        if (_gizmoType != entity->getGizmoType()) {
+            return true;
+        }
+
+        if (_ringProperties != entity->getRingProperties()) {
+            return true;
+        }
+
+        return false;
```
[the remark this answers is not in the document]
Comment (SamGondelman) [reply libraries/entities-renderer/src/RenderableGizmoEntityItem.cpp]: Following a pattern in RenderableEntityItems to make adding properties later easier.

File: libraries/entities-renderer/src/RenderableGizmoEntityItem.cpp [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+                colors << outerEndColor;
+            } else {
+                _solidPrimitive = gpu::TRIANGLE_STRIP;
+                for (float angleRadians = startAtRadians; angleRadians < endAtRadians; angleRadians += SLICE_ANGLE_RADIANS) {
+                    float range = (angleRadians - startAtRadians) / totalRange;
+
+                    points << 0.5f * glm::vec3(cosf(angleRadians) * ringProperties.getInnerRadius(), 0.0f, sinf(angleRadians) * ringProperties.getInnerRadius());
+                    colors << glm::mix(innerStartColor, innerEndColor, range);
+
+                    points << 0.5f * glm::vec3(cosf(angleRadians), 0.0f, sinf(angleRadians));
```
[the remark this answers is not in the document]
Comment (SamGondelman) [reply libraries/entities-renderer/src/RenderableGizmoEntityItem.cpp]: I will do this later!  For now this code was mostly copied from Circle3DOverlay without much change.

File: libraries/entities-renderer/src/RenderableGizmoEntityItem.cpp [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    });
+}
+
+Item::Bound GizmoEntityRenderer::getBound() {
+    auto bound = Parent::getBound();
+    if (_ringProperties.getHasTickMarks()) {
+        glm::vec3 scale = bound.getScale();
+        for (int i = 0; i < 3; i += 2) {
+            if (_ringProperties.getMajorTickMarksLength() + _ringProperties.getInnerRadius() > 1.0f) {
+                scale[i] *= _ringProperties.getMajorTickMarksLength() + _ringProperties.getInnerRadius();
```
[the remark this answers is not in the document]
Comment (SamGondelman) [reply libraries/entities-renderer/src/RenderableGizmoEntityItem.cpp]: Will do later.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16793/HighFidelity-Beta-PR14764-4d83c7f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16793/HighFidelity-Beta-Interface-PR14764-4d83c7f.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16805/HighFidelity-Beta-PR14764-28fab09.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16805/HighFidelity-Beta-PR14764-28fab09.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16805/HighFidelity-Beta-Interface-PR14764-28fab09.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16811/HighFidelity-Beta-PR14764-13cb70a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16811/HighFidelity-Beta-Interface-PR14764-13cb70a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16811/HighFidelity-Beta-PR14764-13cb70a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16811/HighFidelity-Beta-Interface-PR14764-13cb70a.exe

**Comment by SamGondelman** [comment]
build this please

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16820/HighFidelity-Beta-PR14764-13cb70a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16820/HighFidelity-Beta-PR14764-13cb70a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16820/HighFidelity-Beta-Interface-PR14764-13cb70a.dmg

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16823/HighFidelity-Beta-PR14764-13cb70a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16823/HighFidelity-Beta-Interface-PR14764-13cb70a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16823/HighFidelity-Beta-PR14764-13cb70a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16823/HighFidelity-Beta-Interface-PR14764-13cb70a.exe

**Comment by qlilaseguinolaza** [comment]
Currently testing

**Comment by qlilaseguinolaza** [comment]
Currently testing

**Comment by qlilaseguinolaza** [comment]
PR PASS

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16832/HighFidelity-Beta-PR14764-13cb70a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16832/HighFidelity-Beta-Interface-PR14764-13cb70a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16832/HighFidelity-Beta-PR14764-13cb70a.exe

**Comment by SamGondelman** [comment]
BUILD THIS PLEASE

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16844/HighFidelity-Beta-PR14764-13cb70a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16844/HighFidelity-Beta-Interface-PR14764-13cb70a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16844/HighFidelity-Beta-PR14764-13cb70a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16844/HighFidelity-Beta-Interface-PR14764-13cb70a.exe

**Pull request merged**